src/core/maps/transferable_belief_model.h

```diff
@@ -0,0 +1,156 @@
+#ifndef SLAM_CTOR_CORE_TBM_H
+#define SLAM_CTOR_CORE_TBM_H
+
+#include <cassert>
+#include <vector>
+#include <string>
+
+/*------------------ declaration ---------------------*/
+
+//transferable belief model
+class TBM {
+public:
+  //public static belief types
+  static enum class Belief {
```

Comment:
Can we change Belief to private plain enum? Plain enum prevents explicit casts; private visibility should prevent enum-ralated name pollution.

Alternative suggestion is to move enum_class->number conversion to a private constexpr method since static_casts make the code noisy.